contracts/vaults/YearnVaults/RibbonThetaYearnVault.sol

```diff
@@ -0,0 +1,350 @@
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "../../vendor/CustomSafeERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {GnosisAuction} from "../../libraries/GnosisAuction.sol";
+import {
+    OptionsThetaYearnVaultStorage
+} from "../../storage/OptionsVaultYearnStorage.sol";
+import {Vault} from "../../libraries/Vault.sol";
+import {VaultLifecycleYearn} from "../../libraries/VaultLifecycleYearn.sol";
+import {ShareMath} from "../../libraries/ShareMath.sol";
+import {RibbonVault} from "./base/RibbonVault.sol";
+
+contract RibbonThetaYearnVault is RibbonVault, OptionsThetaYearnVaultStorage {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using ShareMath for Vault.DepositReceipt;
+
+    /************************************************
+     *  IMMUTABLES & CONSTANTS
+     ***********************************************/
+
+    // oTokenFactory is the factory contract used to spawn otokens. Used to lookup otokens.
+    address public immutable OTOKEN_FACTORY;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    event OpenShort(
+        address indexed options,
+        uint256 depositAmount,
+        address manager
+    );
+
+    event CloseShort(
+        address indexed options,
+        uint256 withdrawAmount,
+        address manager
+    );
+
+    event NewOptionStrikeSelected(uint256 strikePrice, uint256 delta);
+
+    event PremiumDiscountSet(
+        uint256 premiumDiscount,
+        uint256 newPremiumDiscount
+    );
+
+    event AuctionDurationSet(
+        uint256 auctionDuration,
+        uint256 newAuctionDuration
+    );
+
+    event InstantWithdraw(
+        address indexed account,
+        uint256 amount,
+        uint16 round
+    );
+
+    event InitiateGnosisAuction(
+        address auctioningToken,
+        address biddingToken,
+        uint256 auctionCounter,
+        address manager
+    );
+
+    /************************************************
+     *  CONSTRUCTOR & INITIALIZATION
+     ***********************************************/
+
+    /**
+     * @notice Initializes the contract with immutable variables
+     * @param _weth is the Wrapped Ether contract
+     * @param _usdc is the USDC contract
+     * @param _oTokenFactory is the contract address for minting new opyn option types (strikes, asset, expiry)
+     * @param _gammaController is the contract address for opyn actions
+     * @param _marginPool is the contract address for providing collateral to opyn
+     * @param _gnosisEasyAuction is the contract address that facilitates gnosis auctions
+     */
+    constructor(
+        address _weth,
+        address _usdc,
+        address _oTokenFactory,
+        address _gammaController,
+        address _marginPool,
+        address _gnosisEasyAuction
```

Comment:
We're missing passing the Yearn registry

Reply:
in v1, we had it in constructor because the `asset` parameter was there. but the `asset` parameter is now in initialize, so I pass it in there. `yearnRegistry` is only used once to get the collateral token, so I don't think it is necessary to pass it in constructor, hold it as a constant, and then reference it in initialize. Just pass it into initialize

Reply:
https://github.com/ribbon-finance/ribbon-v2/pull/47/files#diff-93cfb819847795b1c4f019da7b67280790d8b43a53485ac94b47b96d684b3e20R159

Reply:
Understood, sounds good